File: Java/dependency-injection/bastard-injection/src/main/java/com/github/grzesiek_galezowski/BastardInjection/_4_BastardInjection/Core/AcmeProcessingWorkflow.java
package com.github.grzesiek_galezowski.BastardInjection._4_BastardInjection.Core;

interface IProcessingWorkflow
  {
    void SetOutbound(IOutbound outbound);
    void ApplyTo(AcmeMessage message);
  }

  class AcmeProcessingWorkflow implements IProcessingWorkflow
  {
    private final IRepository _repository;
    private final IAuthorization _authorizationRules;
    private IOutbound _outbound;

    public AcmeProcessingWorkflow()
    {
      this(new ActiveDirectoryBasedAuthorization(), new MsSqlBasedRepository());
    }

    //for tests
    private AcmeProcessingWorkflow(
            IAuthorization authorization,
            IRepository repository)
    {
      _authorizationRules = authorization;
      _repository = repository;
    }

    public void SetOutbound(IOutbound outbound)
    {
      _outbound = outbound;
    }

    public void ApplyTo(AcmeMessage message)
    {
      message.AuthorizeUsing(_authorizationRules);
      _repository.Save(message);
      _outbound.Send(message);
    }
  }
}

File: CSharp/DomainSpecificLanguages/DomainSpecificLanguages/AlarmsNestedFunctions/Core/AcmeProcessingWorkflow.cs
using TelecomSystemNestedFunctions.Interfaces;
using TelecomSystemNestedFunctions.Outbound;
using TelecomSystemNestedFunctions.Services;

namespace TelecomSystemNestedFunctions.Core
{
  public interface IAcmeProcessingWorkflow
  {
    void SetOutbound(IOutbound outbound);
    void ApplyTo(AcmeMessage message);
  }

  public class AcmeProcessingWorkflow : IAcmeProcessingWorkflow
  {
    private readonly IRepository _repository;
    private readonly IAuthorization _authorizationRules;
    private IOutbound _outbound;

    public AcmeProcessingWorkflow(
      IAuthorization activeDirectoryBasedAuthorization, 
      IRepository msSqlBasedRepository)
    {
      _authorizationRules = activeDirectoryBasedAuthorization;
      _repository = msSqlBasedRepository;
    }

    public void SetOutbound(IOutbound outbound)
    {
      _outbound = outbound;
    }

    public void ApplyTo(AcmeMessage message)
    {
      message.AuthorizeUsing(_authorizationRules);
      _repository.Save(message);
      _outbound.Send(message);
    }
  }
}